### DISCUSSION TITLE
Filename format for saving images and videos as (year-month-day_hour-minute-second.mp4/jpg/png).

### DISCUSSION THREAD

**Discussion by Trzinka** [blakeblackshear/frigate#7872] [General]
Would it be possible to change the file names of the video and saved images to have the same prefix year-month-day_hour-minute-second.mp4/jpg/png even if the image is captured a few seconds after the object detection is detected and if they could be save everything as mp4, jpg and png in one folder or whatever you have selected for saving. With the current situation, the problem arises when you want to combine all the recorded files in one folder, it is difficult to find which picture belongs to which video.
Here is an example of one vehicle passing 4 cameras.
![Firgate Explorer](https://github.com/blakeblackshear/frigate/assets/40424965/fc596743-12e0-45e4-b6b0-4557b138664f)
Before that, I have to use the program and function as shown in the directory where I save the videos
![Frigat-Sprememba imena](https://github.com/blakeblackshear/frigate/assets/40424965/7eb159d6-2cd4-4e76-84f3-d3ee54d566a0)
![Frigat-Sprememba imena2](https://github.com/blakeblackshear/frigate/assets/40424965/b908f497-e141-4359-8245-de842d767e62)
then find all the folders named after the camera name
![Explorer iskanje map](https://github.com/blakeblackshear/frigate/assets/40424965/dbe63354-fe56-443a-a44e-8d0109b4cd5d)
and move them to another folder
![Explorer premikanje map](https://github.com/blakeblackshear/frigate/assets/40424965/7aacd900-0ea1-4748-bad0-4c7a74c36954)
And again the Advanced Renamer program to add the camera label and then move all the files into a file that I label with months.
The worst thing is that in the end I don't get the desired effect as shown in the first picture.

Of course, I have to praise you for an extraordinary product that convinced me to switch from another NVR program to Frigate.
Thanks.

**Comment by NickM-27** [comment]
> With the current situation, the problem arises when you want to combine all the recorded files in one folder, it is difficult to find which picture belongs to which video.

frigate does not store video in large clips, they are stored in segments which are dynamically used to generate 24/7 recordings or event clips.

Frigate recording segments are considered internal metadata, users should not be interacting with the recordings directly, that is what the webUI and API are for. there are many reasons why storing all of the recording segments into a single folder would be problematic.

I would suggest for both of these you should just use the api to pull a clip.mp4 and the snapshots to save them named whatever you want.

**Comment by Trzinka** [comment]
Dear!

 �

Thanks for the reply. I understand that for normal recording use you don't need to overwrite the files and edit them in a meaningful order. I have a problem with the neighbors who use our lot for parking vehicles and I have started the process in court and the easiest way to prove the case (court allows) is with video and pictures. The court requires me to say for each video clip and picture what I am proving (I need a larger amount of video clips and pictures). With the Ispy NVR program (it has the disadvantage that I can only record movements and not based on the detection of active objects), with which I have a lot of recordings. To conclude, I just asked if there is an option that would make it easier to sort outside of the webUI when dealing with a large amount of recordings.

 �

Greetings!

Robert Perič

Klanec 11

 �

1370 Logatec

+386 41/614-379

Za uporabnike varne poštne:

 ***@***.***> ***@***.***

ali

 ***@***.***> ***@***.***

 �

**Reply by NickM-27** [reply]
Right, I'm suggesting you can use the webui and API to pull clips / snapshots to do this.
